libs/src/sdk/api/tracks/TracksApi.ts

```diff
@@ -66,36 +64,43 @@ export class TracksApi extends TracksApiWithoutStream {
    * Upload a track
    */
   async uploadTrack(requestParameters: UploadTrackRequest) {
-    const { trackFile, coverArtFile, metadata, onProgress } = requestParameters
-
-    // Validate inputs
-    if (!isFileValid(trackFile)) {
-      throw new Error('Track file is not valid')
+    // Parse inputs
+    const result = createUploadTrackSchema().safeParse(requestParameters)
+    if (!result.success) {
+      throw new Error(`uploadTrack arguments not valid: ${result.error}`)
     }
-
-    if (!isFileValid(coverArtFile)) {
-      throw new Error('Cover art file is not valid')
+    const {
+      userId,
+      trackFile,
+      coverArtFile,
+      metadata: parsedMetadata,
+      onProgress
+    } = result.data
+
+    // Transform metadata
+    const metadata = {
+      ...parsedMetadata,
+      ownerId: userId
     }
 
-    if (typeof metadata !== 'object') {
-      throw new Error('Metadata object is not valid')
-    }
+    const isPremium = metadata.isPremium
+    const isUnlisted = metadata.isUnlisted
 
-    const artistId = decodeHashId(requestParameters.artistId)
-    if (artistId === null) {
-      throw new Error('artistId is not valid')
+    // If track is premium, set remixes to false
+    if (isPremium && metadata.fieldVisibility) {
+      metadata.fieldVisibility.remixes = false
     }
 
-    requestParameters.metadata.owner_id = artistId
-
-    const metadataMissingValues = objectMissingValues(
-      requestParameters.metadata,
-      TRACK_REQUIRED_VALUES
-    )
-    if (metadataMissingValues?.length) {
-      throw new Error(
-        `Metadata object is missing values: ${metadataMissingValues}`
-      )
+    // If track is premium or public, set required visibility fields to true
+    if (isPremium || !isUnlisted) {
```

Comment:
Why can we not just do `!isUnlisted`?

Reply:
Good question, we probably can. I pulled this logic from here https://github.com/AudiusProject/audius-client/blob/main/packages/web/src/pages/upload-page/components/EditPage.js#L109 where it only updates the field visibilities to true if it's !unlisted and any of the field visibilities are false. And in the premium case it always sets the field visibilities to true

Let me confirm with saliou that I'm understanding this logic correctly